chef/cookbooks/provisioner/recipes/bootdisk.rb

```diff
@@ -78,13 +96,17 @@
         node[:crowbar_wall][:boot_device] = "disk/by-id/#{bootdisk}"
       end
     end
+
     # If we still have a /dev/disk/by-path name in boot_device here and the
     # devices looks like a virtio device it is safer to use the base device
     # name for bootdisk instead of the by-path name. According to:
     #  https://www.mail-archive.com/systemd-devel@lists.freedesktop.org/msg20607.html
     # the by-path names for virtio device are not stable and don't even exist
-    # anymore with newer udev releases, e.g. on SLES12
-    if File.dirname(node[:crowbar_wall][:boot_device]) == "disk/by-path" and
+    # anymore with newer udev releases
+    # This is only for SLE_11_SP3 sleshammer. On SLE_12 based image
+    # directory "disk/by-path" with virtio disks driver doesn't exist
+    if !node[:crowbar_wall][:boot_device].nil? &&
```

Comment:
I don't think you need to update that test now, since the original `raise` is still there.